linkml_runtime/utils/schemaview.py

```diff
@@ -663,6 +666,54 @@ def expand_curie(self, uri: str) -> str:
                     return ns[pfx] + local_id
         return uri
 
+    @lru_cache(CACHE_SIZE)
+    def get_elements_applicable_by_identifier(self, identifier: str) -> List[str]:
+        """
+        Get a Model element by prefix.
+
+        Parameters
+        ----------
+        identifier: str
+            The identifier in CURIE form
+
+        Returns
+        -------
+        Optional[str]
+                The model element corresponding to the given identifier as available via
+                the id_prefixes mapped to that element.
+
+        """
+        categories = self.get_elements_applicable_by_prefix(self.namespaces().prefix_for(identifier))
+        if len(categories) == 0:
+            logger.warning("no element found for the given curie using id_prefixes attribute"
+                           ": %s, try get_mappings method?", identifier)
+        return categories
+
+    @lru_cache(CACHE_SIZE)
+    def get_elements_applicable_by_prefix(self, prefix: str) -> List[str]:
+        """
+        Get a Model element by prefix.
+
+        Parameters
+        ----------
+        prefix: str
+            The prefix of a CURIE
+
+        Returns
+        -------
+        Optional[str]
+                The model element corresponding to the given prefix as available via
+                the id_prefixes mapped to that element.
+
+        """
+        categories = []
```

Comment:
minor: categories is a biolinkism - use elements instead? not a big deal as this is not exposed to the caller but still good to use internally consistent variable names